src/video/sdl_v.cpp

```diff
@@ -795,29 +769,45 @@ bool VideoDriver_SDL::ChangeResolution(int w, int h)
 	std::unique_lock<std::recursive_mutex> lock;
 	if (_draw_mutex != nullptr) lock = std::unique_lock<std::recursive_mutex>(*_draw_mutex);
 
-	return CreateMainSurface(w, h);
+	return CreateMainSurface(w, h, true);
 }
 
 bool VideoDriver_SDL::ToggleFullscreen(bool fullscreen)
 {
 	std::unique_lock<std::recursive_mutex> lock;
 	if (_draw_mutex != nullptr) lock = std::unique_lock<std::recursive_mutex>(*_draw_mutex);
 
-	_fullscreen = fullscreen;
-	GetVideoModes(); // get the list of available video modes
-	bool ret = !_resolutions.empty() && CreateMainSurface(_cur_resolution.width, _cur_resolution.height);
+	/* Remember current window size */
+	if (fullscreen) {
+		SDL_GetWindowSize(_sdl_window, &_window_size_w, &_window_size_h);
```

Comment:
_window_size_* are only used inside this function, no need for them to be global

Reply:
They don't necessarily need to be global, but they do need to be static: as it's currently set up, these variables need to exist outside the scope of this function for remembering the value of the window size throughout multiple calls of this function. Of course, this could all possibly be refactored, as maybe it's not clear enough.